### ISSUE TITLE
Feature request for deskop and mobile a-sky zoom-in and zoom-out

### ISSUE THREAD

**Issue by mybmgithub** [tizzle/aframe-orbit-controls-component#20]
First off, thank you author for the aframe-orbit-controls-component repo. 

My feature request is for _desktop_ and _mobile_ browser _mousewheel_ and _pinch and zoom_ support for zooming on the `<a-sky>` element.

Example:
I want to be able center `<a-camera position=" 0 0 0" zoom="[value_here]"></a-camera>` 
This way I can zoom-in or zoom-out on any area of say a room that is set on the `<a-sky src=""></a-sky>`.

Thank you.

**Comment by tizzle** [comment]
Hey @mybmgithub,

the orbit-controls are not intended to cover this use case. This lies in the basic idea the orbit controls are built around: You orbit a target object. This means the camera is moving around a target on a circular path, always looking at the target. Say you have a target object at [0,0,0]. When you zoom in the camera it moves closer to [0,0,0], when you zoom out the camera moves farther away. It's important to know that the camera cannot be any closer to [0,0,0] than [0,0,0].

In he use-case you are describing this leads to a problem that cannot be solved with orbit-controls. You use a sky object, that is essentially just a sphere with a radius so big that it encompasses all objects in the scene, including the camera. This sphere has you panoramic texture applied on the inside so it feels like you are in a room. The sky is initially placed at [0,0,0], meaning the center of the sphere is at that position. 

When you use the orbit-controls with this setup and have a target at [0,0,0] you won't be able to get the camera any closer to the sky sphere than zooming it in until it's at [0,0,0] which depending on the size of your sky might be quite far away. Due to the inner workings of orbit-controls—and the purpose it was built for—you won't be able to overcome this. 

I suggest you look for other components that fit your needs better or start writing your own component that does exactly what you want to achieve. The docs on the Aframe website have a [chapter](https://aframe.io/docs/0.5.0/core/component.html) on how to get started.

Sorry to be of little help with this.

Cheers

**Comment by mybmgithub** [comment]
Thank you tizzle for your follow-up. I will continue my search for a package that can achieve this goal or roll my own solution in the end.

**Comment by tizzle** [comment]
Hey @mybmgithub,

have you had luck finding something that suits your needs? 

Cheers

Till

**Comment by mybmgithub** [comment]
I have not found anything else that can do this for our project. Nor have I written a Component. But, I did find a fix for another major need of pitching up and down in our project while on desktop or mobile. And for zoom, I may settle (for now) on the zoom property from the Camera and put a + and - in the UI to allow for zooming to happen. I tested zoom on the Camera and that is what I need. So, that would at least give the app the zoom feature it really should have.
